apps/managedidentity/managedidentity_test.go

```diff
@@ -217,6 +263,271 @@ func Test_SystemAssigned_Returns_Token_Success(t *testing.T) {
+
+	mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusUnauthorized),
+		mock.WithHTTPHeader(headers),
+	)
+	setCustomAzureArcPlatformPath(t, "")
+
+	client, err := New(SystemAssigned(), WithHTTPClient(&mockClient))
+	if err != nil {
+		t.Fatalf(`no errors were expected, got : %v"`, err)
+	}
+	result, err := client.AcquireToken(context.Background(), resource)
+	if err == nil || !strings.Contains(err.Error(), "platform not supported") {
+		t.Fatalf(`expected error: "%v" got error: "%v"`, "platform not supported", err)
+
+	}
+	if result.AccessToken != "" {
+		t.Fatalf("access token should be empty")
+	}
+}
+
+func TestAzureArcErrors(t *testing.T) {
+	setEnvVars(t, AzureArc)
+	setCustomAzureArcFilePath(t, fakeAzureArcFilePath)
+	testCaseFilePath := getMockFilePath(t)
+	testCases := []struct {
+		name          string
+		headerValue   string
+		expectedError string
+		fileSize      int64
+	}{
+		{
+			name:          "No www-authenticate header",
+			expectedError: "response has no www-authenticate header",
+		},
+		{
+			name:          "Basic realm= not found",
+			headerValue:   "Basic ",
+			expectedError: "basic realm= not found in the string, instead found: Basic ",
+		},
+		{
+			name:          "Invalid file extension",
+			headerValue:   "Basic realm=/path/to/secret.txt",
+			expectedError: "invalid file extension, expected .key, got .txt",
+		},
+		{
+			name:          "Invalid file path",
+			headerValue:   "Basic realm=" + filepath.Join("path", "to", "secret.key"),
+			expectedError: "invalid file path, expected " + testCaseFilePath + ", got " + filepath.Join("path", "to"),
+		},
+		{
+			name:          "Unable to get file info",
+			headerValue:   basicRealm + filepath.Join(testCaseFilePath, wrongSecretKey),
+			expectedError: "failed to get metadata",
+		},
+		{
+			name:          "Invalid secret file size",
+			headerValue:   basicRealm + filepath.Join(testCaseFilePath, secretKey),
+			expectedError: "invalid secret file size, expected 4096, file size was 5000",
+			fileSize:      5000,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(string(testCase.name), func(t *testing.T) {
+			mockClient := mock.Client{}
+			mockFilePath := filepath.Join(testCaseFilePath, secretKey)
+			setCustomAzureArcPlatformPath(t, testCaseFilePath)
+			createMockFile(t, mockFilePath, testCase.fileSize)
+			headers := http.Header{}
+			headers.Set(wwwAuthenticateHeaderName, testCase.headerValue)
+
+			mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusUnauthorized),
+				mock.WithHTTPHeader(headers),
+			)
+
+			responseBody, err := getSuccessfulResponse(resource)
+			if err != nil {
+				t.Fatalf(errorFormingJsonResponse, err.Error())
+			}
+
+			mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK), mock.WithHTTPHeader(headers),
+				mock.WithBody(responseBody))
+
+			client, err := New(SystemAssigned(), WithHTTPClient(&mockClient))
+			if err != nil {
+				t.Fatalf(`expected error: "%v" "`, testCase.expectedError)
```

Comment:
1. when failing due to an error, include that error
2. you expect an error from `AcquireToken`, not `New`
```suggestion
				t.Fatal(err)
```